### PR TITLE
Audio element

### PR DESCRIPTION
**Pull request by tpatel**
- Add Audio element
- Enabled automated mime type detection

### CODE REVIEW

File: src/chainlit/frontend/src/components/element/audio.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  element: IAudioElement;
+}
+
+export default function AudioElement({ element }: Props) {
+  if (!element.url && !element.content) {
+    return null;
+  }
+  const className = `${element.display}-audio`;
+  const src = element.url || URL.createObjectURL(new Blob([element.content!]));
+  return <audio controls src={src} className={className}></audio>;
```
Comment (willydouhard) [inline src/chainlit/frontend/src/components/element/audio.tsx:13]: Is there a way to display the name of the audio file in the native audio component? Or should we add it ourselves?

File: src/chainlit/frontend/src/state/element.ts [hunk 2]
```diff
@@ -42,6 +49,12 @@ export interface IPdfElement extends IElement {
   url?: string;
 }
 
+export interface IAudioElement extends IElement {
+  type: 'audio';
+  content?: string;
```
Comment (willydouhard) [inline src/chainlit/frontend/src/state/element.ts:54]: I don't think the content should be of type string, maybe you copy pasted from the TextElement?

**Review by willydouhard — CHANGES_REQUESTED** [review]
looks good! minor things to change

File: cypress/e2e/audio_element/spec.cy.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    cy.wait(["@settings"]);
+  });
+
+  it("should be able to display an audio element", () => {
+    cy.get("#welcome-screen").should("exist");
+
+    cy.get(".message").should("have.length", 1);
+    cy.get(".message").eq(0).find(".inline-audio").should("have.length", 1);
+  });
+});
```
Comment (willydouhard) [inline cypress/e2e/audio_element/spec.cy.ts:14]: [this](https://github.com/cypress-io/cypress/issues/1750) looks interesting to actually test the audio

**Review by willydouhard — CHANGES_REQUESTED** [review]

File: src/chainlit/frontend/src/components/element/audio.tsx (hunk 1 above)
Comment (tpatel) [reply src/chainlit/frontend/src/components/element/audio.tsx:13]: Good point! Here is how it looks:

image

I took some inspiration from this part of the figma designs:

image

File: src/chainlit/frontend/src/state/element.ts (hunk 2 above)
Comment (tpatel) [reply src/chainlit/frontend/src/state/element.ts]: fixed

File: src/chainlit/frontend/src/components/element/audio.tsx [hunk 4]
```diff
@@ -1,14 +1,31 @@
...
 
 export default function AudioElement({ element }: Props) {
+  const theme = useTheme();
   if (!element.url && !element.content) {
     return null;
   }
   const className = `${element.display}-audio`;
   const src = element.url || URL.createObjectURL(new Blob([element.content!]));
-  return <audio controls src={src} className={className}></audio>;
+  return (
```
Comment (willydouhard) [inline src/chainlit/frontend/src/components/element/audio.tsx:16]: it is better to use the Typography element here like 

```jsx
      <Typography
        color="text.primary"
        sx={{ fontSize: '14px', fontWeight: 700 }}
      >...</Typography>
```

unless the color is different from the main or secondary color in the theme?

File: src/chainlit/frontend/src/components/element/audio.tsx (hunk 1 above)
Comment (willydouhard) [reply src/chainlit/frontend/src/components/element/audio.tsx]: nice!

File: src/chainlit/frontend/src/components/element/audio.tsx (hunk 4 above)
Comment (tpatel) [reply src/chainlit/frontend/src/components/element/audio.tsx:16]: I've used a color that's different from the main/secondary color. It's slightly muted, like the Figma design of the image filename (that's the closest thing I've found for this purpose).

image

File: cypress/e2e/audio_element/spec.cy.ts (hunk 3 above)
Comment (tpatel) [reply cypress/e2e/audio_element/spec.cy.ts:21]: I've found a way to load the audio file, wait for it to be loaded and then check that it was loaded correctly ✅

**Review by willydouhard — APPROVED** [review]

**Pull request merged**